// leather/dist/index.mjs
import { lazystream } from './util.mjs';
import { attributes as attributes$1 } from './extractors/bmp.mjs';
import { attributes as attributes$2 } from './extractors/dds.mjs';
import { attributes as attributes$3 } from './extractors/gif.mjs';
import { attributes as attributes$4 } from './extractors/icns.mjs';
import { attributes as attributes$5 } from './extractors/ico.mjs';
import { attributes as attributes$6 } from './extractors/j2c.mjs';
import { attributes as attributes$7 } from './extractors/jpg.mjs';
import { attributes as attributes$8 } from './extractors/ktx.mjs';
import { attributes as attributes$9 } from './extractors/png.mjs';
import { attributes as attributes$a } from './extractors/pnm.mjs';
import { attributes as attributes$b } from './extractors/psd.mjs';
import { attributes as attributes$c } from './extractors/svg.mjs';
import { attributes as attributes$d } from './extractors/tiff.mjs';
import { attributes as attributes$e } from './extractors/webp.mjs';
import { attributes as attributes$f } from './extractors/xpm.mjs';
import { attributes as attributes$g } from './extractors/xbm.mjs';
import { attributes as attributes$j } from './extractors/fit.mjs';
import { attributes as attributes$h } from './extractors/cel.mjs';
import { attributes as attributes$i } from './extractors/hdr.mjs';
import { attributes as attributes$k } from './extractors/avi.mjs';
import { attributes as attributes$l } from './extractors/fli.mjs';
import { attributes as attributes$m } from './extractors/flv.mjs';
import { attributes as attributes$n } from './extractors/ogv.mjs';
import { attributes as attributes$o } from './extractors/mp4.mjs';
import { attributes as attributes$p } from './extractors/webm.mjs';
import { attributes as attributes$q } from './extractors/wmv.mjs';

// extractor mapping table based on `lazystream(file).identifier()`
const extractors = {
    bmp: attributes$1,
    dds: attributes$2,
    gif: attributes$3,
    icns: attributes$4,
    ico: attributes$5, // also used for: cur
    j2c: attributes$6, // also used for: jp2
    jpg: attributes$7,
    ktx: attributes$8,
    png: attributes$9,
    pnm: attributes$a,
    psd: attributes$b,
    svg: attributes$c,
    tiff: attributes$d,
    webp: attributes$e,
    xpm: attributes$f,
    xbm: attributes$g,
    cel: attributes$h,
    hdr: attributes$i,
    fit: attributes$j,
    avi: attributes$k,
    fli: attributes$l, // also used for: flc
    flv: attributes$m,
    ogv: attributes$n,
    mp4: attributes$o, // also used for: m4v
    webm: attributes$p, // also used for: mkv
    wmv: attributes$q,
};

function attributes(file) {
    const stream = lazystream(file);
    const result = {width: 0, height: 0, size: 0, mime: undefined};
    const extract = extractors[stream.identifier()];

    if (extract) Object.assign(result, extract(stream));

    stream.close();

    return result;
}

export { attributes };
