// leather/dist/index.cjs
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var util = require('./util.cjs');
var bmp = require('./extractors/bmp.cjs');
var dds = require('./extractors/dds.cjs');
var gif = require('./extractors/gif.cjs');
var icns = require('./extractors/icns.cjs');
var ico = require('./extractors/ico.cjs');
var j2c = require('./extractors/j2c.cjs');
var jpg = require('./extractors/jpg.cjs');
var ktx = require('./extractors/ktx.cjs');
var png = require('./extractors/png.cjs');
var pnm = require('./extractors/pnm.cjs');
var psd = require('./extractors/psd.cjs');
var svg = require('./extractors/svg.cjs');
var tiff = require('./extractors/tiff.cjs');
var webp = require('./extractors/webp.cjs');
var xpm = require('./extractors/xpm.cjs');
var xbm = require('./extractors/xbm.cjs');
var fit = require('./extractors/fit.cjs');
var cel = require('./extractors/cel.cjs');
var hdr = require('./extractors/hdr.cjs');
var avi = require('./extractors/avi.cjs');
var fli = require('./extractors/fli.cjs');
var flv = require('./extractors/flv.cjs');
var ogv = require('./extractors/ogv.cjs');
var mp4 = require('./extractors/mp4.cjs');
var webm = require('./extractors/webm.cjs');
var wmv = require('./extractors/wmv.cjs');

// extractor mapping table based on `lazystream(file).identifier()`
const extractors = {
    bmp: bmp.attributes,
    dds: dds.attributes,
    gif: gif.attributes,
    icns: icns.attributes,
    ico: ico.attributes, // also used for: cur
    j2c: j2c.attributes, // also used for: jp2
    jpg: jpg.attributes,
    ktx: ktx.attributes,
    png: png.attributes,
    pnm: pnm.attributes,
    psd: psd.attributes,
    svg: svg.attributes,
    tiff: tiff.attributes,
    webp: webp.attributes,
    xpm: xpm.attributes,
    xbm: xbm.attributes,
    cel: cel.attributes,
    hdr: hdr.attributes,
    fit: fit.attributes,
    avi: avi.attributes,
    fli: fli.attributes, // also used for: flc
    flv: flv.attributes,
    ogv: ogv.attributes,
    mp4: mp4.attributes, // also used for: m4v
    webm: webm.attributes, // also used for: mkv
    wmv: wmv.attributes,
};

function attributes(file) {
    const stream = util.lazystream(file);
    const result = {width: 0, height: 0, size: 0, mime: undefined};
    const extract = extractors[stream.identifier()];

    if (extract) Object.assign(result, extract(stream));

    stream.close();

    return result;
}

exports.attributes = attributes;
